unitex/src/fr/umlv/unitex/graphtools/FindAndReplace.java

```diff
@@ -396,4 +415,212 @@ public static String checkReplaceAll(ArrayList<GenericGraphBox> boxes, String se
+        }
+        break;
+    }
+    return findNextInSeq(zone, seq, box.getTransitions().get(0), i + 1, highlight, caseSensitive, useRegex);
+  }
+
+  /**
+   * Returns true if and only if this box and the following boxes match the sequence.
+   *
+   * @param graphicalZone the GraphicalZone containing the sequence.
+   * @param findSeqList the list containing the sequence to search for.
+   * @param genericGraphBox the first box of the sequence.
+   * @param highlight true if the sequence must be highlighted, false otherwise.
+   * @param caseSensitive true if the search must be case sensitive, false otherwise.
+   * @param useRegex true if the search must use regular expressions, false otherwise.
+     * @return true if the box and the following boxes match the sequence, false otherwise.
+     */
+  public static boolean isSeq(GraphicalZone graphicalZone, ArrayList<String> findSeqList, GenericGraphBox
+    genericGraphBox, boolean highlight, boolean caseSensitive, boolean useRegex) {
+    if(findSeqList.size() == 1 && boxEquals(findSeqList.get(0), genericGraphBox.getContent(), caseSensitive, useRegex)) {
+      if (highlight) {
+        //genericGraphBox.setSelected(highlight);
+        genericGraphBox.setHighlight(true);
+        graphicalZone.getSelectedBoxes().add(genericGraphBox);
+        graphicalZone.setHighlight(genericGraphBox, true);
+      }
+      return true;
+    }
+    if (boxEquals(findSeqList.get(0), genericGraphBox.getContent(), caseSensitive, useRegex) && genericGraphBox
+      .getTransitions().size() == 1) {
+      if (highlight) {
+        //genericGraphBox.setSelected(highlight);
+        genericGraphBox.setHighlight(true);
+        graphicalZone.getSelectedBoxes().add(genericGraphBox);
+        graphicalZone.setHighlight(genericGraphBox, true);
+      }
+      if (!findNextInSeq(graphicalZone, findSeqList, genericGraphBox.getTransitions().get(0), 1, highlight,
+        caseSensitive, useRegex)) {
+        graphicalZone.unSelectAllBoxes();
+        graphicalZone.setHighlight(false);
+        graphicalZone.removeHighlight();
+        return false;
+      }
+      return true;
+    }
+    graphicalZone.setHighlight(false);
+    graphicalZone.unSelectAllBoxes();
+    graphicalZone.removeHighlight();
+    return false;
+  }
+
+  /**
+   * Replace all sequence with replaceSeqList and returns the number of sequences
+   * which have been replaced.
+   *
+   * @param zone the GraphicalZone containing the sequence.
+   * @param findSeqList the list containing the sequence to search for.
+   * @param replaceSeqList the list containing the sequence to replace with.
+   * @param caseSensitive true if the search must be case sensitive, false otherwise.
+   * @param useRegex true if the search must use regular expressions, false otherwise.
+     * @return the number of sequences which have been replaced.
+     */
+  public static int replaceAllSeq(GraphicalZone zone, ArrayList<String> findSeqList, ArrayList<String>
```

Comment:
I would combine the 4 if conditions into a single if else ladder because the conditions are exclusive